File: 3451-string-compression-iii/3451-string-compression-iii.java
class Solution {
    public String compressedString(String word) {
        StringBuilder comp = new StringBuilder();
        int i=0;
        while(i < word.length()){
            int cnt = 0;
            char currChar = word.charAt(i);
            while(i < word.length() && cnt < 9 && word.charAt(i)==currChar){
                i++;
                cnt++;
            }
            comp.append(cnt).append(currChar);
        }
        return comp.toString();
    }
}

File: 3451-string-compression-iii/3451-string-compression-iii.cpp
class Solution {
public:
    string compressedString(string word) {
        string comp = "";
        int cnt = 1;
        for(int i=1; i<word.size(); i++){
            if(word[i] == word[i-1]){
                cnt++;
                if(cnt == 10){
                    comp += "9" + string(1, word[i-1]);
                    cnt = 1;
                }
            }
            else{
                comp += to_string(cnt) + word[i-1];
                cnt = 1;
            }
        }
        comp += to_string(cnt)+word[word.size()-1];
        return comp;
    }
};